components/gfx_traits/print_tree.rs

```diff
@@ -22,20 +22,29 @@ impl PrintTree {
         }
     }
 
-    /// Descend one level in the tree with the given title.
-    pub fn new_level(&mut self, title: String) {
+    /// Descend one level in the tree with the given items string.
+    pub fn new_level(&mut self, queued_item: String) {
         self.flush_queued_item("\u{251C}\u{2500}");
 
         self.print_level_prefix();
-        println!("\u{251C}\u{2500} {}", title);
+
+        let items: Vec<&str> = queued_item.split("\n").collect();
+        println!("\u{251C}\u{2500} {}", items[0]);
+        for i in 1..items.len() {
+            self.print_level_child_prefix();
+            print!("{}", items[i]);
+            if i != items.len() {
```

Comment:
I would probably use `i < items.len()`, because if you have a bug in the `for` loop and `i` becomes bigger then `items.len()`, you won't print some unwanted `'\n'`